crates/transpiler/src/passes/consolidate_blocks.rs

```diff
@@ -352,7 +442,189 @@ pub fn run_consolidate_blocks(
+                ])),
+            )
+            .expect("Error while adding SXGate to target");
+        target
+            .add_instruction(
+                StandardGate::X.into(),
+                &[],
+                None,
+                Some(IndexMap::from_iter([
+                    (
+                        Qargs::Concrete(smallvec![PhysicalQubit(0)]).to_owned(),
+                        None,
+                    ),
+                    (
+                        Qargs::Concrete(smallvec![PhysicalQubit(1)]).to_owned(),
+                        None,
+                    ),
+                ])),
+            )
+            .expect("Error while adding XGate to target");
+        target
+            .add_instruction(
+                StandardGate::RZ.into(),
+                &[Param::ParameterExpression(phi)],
+                None,
+                Some(IndexMap::from_iter([
+                    (
+                        Qargs::Concrete(smallvec![PhysicalQubit(0)]).to_owned(),
+                        None,
+                    ),
+                    (
+                        Qargs::Concrete(smallvec![PhysicalQubit(1)]).to_owned(),
+                        None,
+                    ),
+                ])),
+            )
+            .expect("Error while adding RXGate to target");
+        target
+            .add_instruction(
+                StandardGate::CZ.into(),
+                &[],
+                None,
+                Some(IndexMap::from_iter([
+                    (
+                        Qargs::Concrete(smallvec![PhysicalQubit(0), PhysicalQubit(1)]).to_owned(),
+                        None,
+                    ),
+                    (
+                        Qargs::Concrete(smallvec![PhysicalQubit(1), PhysicalQubit(0)]).to_owned(),
+                        None,
+                    ),
+                ])),
+            )
+            .expect("Error while adding CZGate to target");
+
+        // Convert the circuit to a DAG.
+        let mut circ_as_dag =
+            DAGCircuit::from_circuit_data(&circuit, false, None, None, None, None)
+                .expect("Error converting circuit to DAG.");
+        // Run the pass
+        run_consolidate_blocks(&mut circ_as_dag, true, None, Some(&target))
+            .expect("Error while running the consolidate blocks pass.");
+
+        let circ_result = dag_to_circuit(&circ_as_dag, false)
+            .expect("Error while converting the DAG to a circuit.");
+
+        if let Some(op) = circ_result.iter().next() {
+            if op.op.name() != "unitary" {
+                panic!("The pass did not correctly extract into a unitary");
+            }
+            if circ_result.__len__() != 1 {
+                panic!(
+                    "The resulting circuit from our run did not result in one single unitary gate."
+                );
+            }
+        } else {
+            panic!("The resulting circuit was unexpectedly empty.");
+        };
```

Comment:
I think for clarity and more thorough checking I'd write the test like:

```suggestion
        let data = circuit_result.data();
        if data.len() != 1 {
            panic!("The output circuit had {data.len()} instructions but there should only be 1 unitary gate");
        }
        let OperationRef::Unitary(unitary) = data.first().unwrap().op.view() else {
            panic!("The output circuit didn't result in a single unitary gate");
        }
        let expected = aview2([
            [...],
            [...],
        ])
        assert!(abs_diff_eq(unitary.matrix_view(), expected));
```
Check my syntax here since the github webui is where syntax errors are injected and also filling in the matrix with the expected value.

Reply:
I ended up simplifying the test to just an identity cancellations as we discussed before.